src/Marten/Linq/Selectors/SerializationSelector.cs

```diff
@@ -15,14 +15,14 @@ public SerializationSelector(ISerializer serializer)
 
         public T Resolve(DbDataReader reader)
         {
-            using var json = reader.GetTextReader(0);
+            using var json = reader.GetStream(0);
             return _serializer.FromJson<T>(json);
         }
 
-        public Task<T> ResolveAsync(DbDataReader reader, CancellationToken token)
+        public async Task<T> ResolveAsync(DbDataReader reader, CancellationToken token)
```

Comment:
And let's look to see about using ValueTask. Might also want the actual reading from the Stream to be async. We should look at how the ASP.net JsonFormatter uses STJ and see if it buffers data from the Stream or anything like that. Not saying this is necessarily wrong, but we should do our homework.

Reply:
If it's the right one - then it looks like is doing that the same way as we do: https://github.com/dotnet/aspnetcore/blob/main/src/Mvc/Mvc.Core/src/Formatters/SystemTextJsonOutputFormatter.cs#L77 (plus they're doing additional serialisation from object that we don't want to do).

Reply:
I think that we should add method `ToStream` and `ToStreamAsync` to `IQueryable` etc. Plus some custom output formatter that'll just return stream with proper encoding.